test/Microsoft.SqlTools.ServiceLayer.UnitTests/Utility/SqlScriptFormatterTests.cs

```diff
@@ -308,6 +310,55 @@ public void GuidTest()
 
         #endregion
 
+        #region DecodeMultipartIdentifier Tests
+
+        [Theory]
+        [MemberData(nameof(DecodeMultipartIdentifierTestData))]
+        public void DecodeMultipartIdentifierTest(string input, string[] output)
+        {
+            // If: I decode the input
+            string[] decoded = SqlScriptFormatter.DecodeMultipartIdenfitier(input);
+
+            // Then: The output should match what was expected
+            Assert.Equal(output, decoded);
+        }
+
+        public static IEnumerable<object> DecodeMultipartIdentifierTestData
+        {
+            get
+            {
+                yield return new object[] {"identifier", new[] {"identifier"}};
+                yield return new object[] {"simple.split", new[] {"simple", "split"}};
+                yield return new object[] {"multi.simple.split", new[] {"multi", "simple", "split"}};
+                yield return new object[] {"[escaped]", new[] {"escaped"}};
+                yield return new object[] {"[escaped].[split]", new[] {"escaped", "split"}};
+                yield return new object[] {"[multi].[escaped].[split]", new[] {"multi", "escaped", "split"}};
+                yield return new object[] {"[escaped]]characters]", new[] {"escaped]characters"}};
+                yield return new object[] {"[multi]]escaped]]chars]", new[] {"multi]escaped]chars"}};
+                yield return new object[] {"[multi]]]]chars]", new[] {"multi]]chars"}};
+                yield return new object[] {"unescaped]chars", new[] {"unescaped]chars"}};
+                yield return new object[] {"multi]unescaped]chars", new[] {"multi]unescaped]chars"}};
+                yield return new object[] {"multi]]chars", new[] {"multi]]chars"}};
+                yield return new object[] {"[escaped.dot]", new[] {"escaped.dot"}};
+                yield return new object[] {"mixed.[escaped]", new[] {"mixed", "escaped"}};
+                yield return new object[] {"[escaped].mixed", new[] {"escaped", "mixed"}};
```

Comment:
Although I wouldn't recommend it `SELECT dbo.[[].weird FROM [[]` totally works. I will add a case for that.